packages/wallet-standard/src/register.ts

```diff
@@ -1,14 +1,71 @@
-import type { WalletsWindow } from "@wallet-standard/base";
-import { GlowWallet } from "./wallet.js";
-
-declare const window: WalletsWindow;
-
-export function register(): void {
-  try {
-    (window.navigator.wallets ||= []).push(({ register }) =>
-      register(new GlowWallet())
-    );
-  } catch (error) {
-    console.error(error);
-  }
+// This is copied from @wallet-standard/wallet
+
+import type {
+    DEPRECATED_WalletsWindow,
+    Wallet,
+    WalletEventsWindow,
+    WindowRegisterWalletEvent as WindowRegisterWalletEventInterface,
+    WindowRegisterWalletEventCallback,
+} from '@wallet-standard/base';
+
+export function registerWallet(wallet: Wallet): void {
+    const callback: WindowRegisterWalletEventCallback = ({ register }) => register(wallet);
+    try {
+        (window as WalletEventsWindow).dispatchEvent(new WindowRegisterWalletEvent(callback));
+    } catch (error) {
+        console.error('wallet-standard:register-wallet event could not be dispatched\n', error);
+    }
+    try {
+        (window as WalletEventsWindow).addEventListener('wallet-standard:app-ready', ({ detail: api }) =>
+            callback(api)
+        );
+    } catch (error) {
+        console.error('wallet-standard:app-ready event listener could not be added\n', error);
+    }
```

Comment:
This is the focus of this update. We are no longer using the global `window.navigator.wallets` interface, but instead using the window's built-in event listener queue. This prevents wallets from directly manipulating the array before the app loads, and avoids conflicts with future window globals.